File: Object Oriented Programming/learning/classes_and_objects/exercises/py/ex-2.py
"""
Assign and print the roll number, phone number and address of two students 
having names "Sam" and "John" respectively by creating two objects of the class 'Student'.
"""

# define class 
class Student:
    # initialize constructor
    def __init__(
            self, 
            name = "",
            roll_no = 0,
            address = "",
            phone_number = ""

    ):
        self.name = name
        self.roll = roll_no
        self.address = address
        self.phone = phone_number

# create objects and assign values
std1 = Student("Sam", 1, "AC204", "02439884")
std2 = Student("John", 2, "AC205", "0544148216")

# print values
print(f"Student 1:\n"
      f"Name: {std1.name}\n"
      f"Roll number: {std1.roll}\n"
      f"Adress: {std1.address}\n"
      f"Phone number: {std1.phone}\n")

print(f"Student 2:\n"
      f"Name: {std2.name}\n"
      f"Roll number: {std2.roll}\n"
      f"Adress: {std2.address}\n"
      f"Phone number: {std2.phone}\n")

File: Object Oriented Programming/learning/classes_and_objects/exercises/cpp/ex-2.cpp
/*
Assign and print the roll number, phone number and address of two students 
having names "Sam" and "John" respectively by creating two objects of the class 'Student'.
*/

#include <iostream>
#include <string>
using namespace std;

// create class
class Student {
    public:
        int roll_no;
        string name, address, phone_number;

    // constructor to initialize objects
    Student(string n, string a, string p, int r) {
        name = n;
        address = a;
        phone_number = p;
        roll_no = r;
    }
};

int main() {
    // create objects of class
    Student std1(
        "Sam", "AC204", "0243988529", 2
    );

    Student std2(
        "John", "AC205", "0544148216", 3
    );

    // Print student information
    cout << "Student 1:\n";
    cout << "Name: " << std1.name << endl;
    cout << "Roll No: " << std1.roll_no << endl;
    cout << "Address: " << std1.address << endl;
    cout << "Phone Number: " << std1.phone_number << endl;

    cout << "\nStudent 2:\n";
    cout << "Name: " << std2.name << endl;
    cout << "Roll No: " << std2.roll_no << endl;
    cout << "Address: " << std2.address << endl;
    cout << "Phone Number: " << std2.phone_number << endl;

    return 0;
}